game-service/src/main/java/org/libertybikes/game/core/GameRound.java

```diff
@@ -149,6 +149,45 @@ public void updatePlayerDirection(Session playerSession, InboundMessage msg) {
         c.player.ifPresent((p) -> p.setDirection(msg.direction));
     }
 
+    public boolean addExternalAi(Session s, String playerId, String playerName) {
```

Comment:
this method is the exact same as `addPlayer` except for 1 line. We should re-use the `addPlayer` method instead of duplicating all this code. Does the game-service need to know if it's talking to an AI vs. a real player? I think it could treat both the same way because it still sends the game board and receives direction updates the same way in both cases.